handler.go

```diff
@@ -194,5 +194,25 @@ func (pset pathConfigSet) find(path string) (pc *pathConfig, subpath string) {
 	if i > 0 && strings.HasPrefix(path, pset[i-1].path+"/") {
 		return &pset[i-1], path[len(pset[i-1].path)+1:]
 	}
-	return nil, ""
+
+	// Otherwise now look for the shortest prefix pathConfig
```

Comment:
Don't you want the longest prefix?  Test case I'm thinking of:

["/", "/abc/", "/abc/def/", "/xyz/"] for "/abc/foo".  I would expect it to match "/abc/", not "/".

Reply:
Thanks for pointing that out @zombiezen, in deed, I used the wrong wording, yes it is longest prefix, articulated, but I meant to say shortest prefix after matching against the "/abc/foo" e.g

Route|Prefix stripped from Query i.e. stripPrefix(query, route)|length of strip
---|---|---
/| abc/def|7
/abc/|def|3

Giving closest match of "/abc/".

I'll change this to longest prefix :)

Reply:
Just the name I mean, the code already does that.